nin/frontend/app/scripts/components/main.tsx

```diff
@@ -183,15 +191,14 @@ class Main extends React.Component {
         }
 
         const globalJSErrors = Object.assign({}, this.state.globalJSErrors);
-        delete globalJSErrors[event.type];
+        delete globalJSErrors[event.name];
         this.setState({globalJSErrors});
       } catch (e) {
         e.context = "WS load of " + event.name + " failed";
-        e.type = event.type;
         e.name = event.name;
 
         const globalJSErrors = Object.assign({}, this.state.globalJSErrors);
-        globalJSErrors[event.type] = e;
+        globalJSErrors[event.name] = e;
```

Comment:
changing from type to kind vs name here. What impact does this have on the behavior?